snuba/request/__init__.py

```diff
@@ -3,16 +3,18 @@
 from collections import ChainMap
 from dataclasses import dataclass
 from deprecation import deprecated
-from typing import Any, Mapping
+from typing import Any, Mapping, Union
 
 from snuba.query.logical import Query
+from snuba.query.composite import CompositeQuery
+from snuba.query.data_source.simple import Entity
 from snuba.request.request_settings import RequestSettings
 
 
 @dataclass(frozen=True)
 class Request:
     id: str
-    query: Query
+    query: Union[Query, CompositeQuery[Entity]]
```

Comment:
There are a bunch of places where this attribute is referenced and assume it is a `Query`. I suspect several of them would not type check now.

```
(snuba) ~/snuba  (feat/add_basic_classifier *)$ git grep request\.query
snuba/datasets/dataset.py:        if isinstance(request.query, Query):
snuba/datasets/dataset.py:            entity = get_entity(request.query.get_from_clause().key)
snuba/datasets/dataset.py:            return CompositeExecutionPipeline(request.query, request.settings, runner)
snuba/pipeline/pipeline_delegator.py:        return executor.execute(self.__request.query)
snuba/pipeline/simple_pipeline.py:            request, runner, self.build_planner(request.query, request.settings),
snuba/querylog/__init__.py:        final = str(request.query.get_final())
snuba/web/query.py:    query_entity = request.query.get_from_clause()
snuba/web/query.py:        request.query.get_sample() is not None and request.query.get_sample() != 1.0
snuba/web/query.py:                request.query, request.extensions[name], request.settings
snuba/web/query.py:        request.query.set_final(False)
tests/query/test_query_ast.py:        .build_planner(request.query, request.settings)
```

Did you verify that? If not, could you check they type check and fix them if not ?

Reply:
I went through them and fixed any that were broken.